cocotb/share/lib/gpi_log/gpi_logging.cpp

```diff
@@ -1,174 +1,125 @@
-    const char *name,
-    int level,
-    const char *pathname,
-    const char *funcname,
-    long lineno,
-    const char *msg,
-    va_list argp)
-{
+extern "C" void gpi_vlog(const char *name, int level, const char *pathname,
+                         const char *funcname, long lineno, const char *msg,
+                         va_list argp) {
     if (current_handler) {
-        (*current_handler)(
-            current_userdata,
-            name,
-            level,
-            pathname,
-            funcname,
-            lineno,
-            msg,
-            argp);
+        (*current_handler)(current_userdata, name, level, pathname, funcname,
+                           lineno, msg, argp);
     } else {
-        gpi_native_logger_vlog(name,
-            level,
-            pathname,
-            funcname,
-            lineno,
-            msg,
-            argp);
+        gpi_native_logger_vlog(name, level, pathname, funcname, lineno, msg,
+                               argp);
     }
 }
 
-
-extern "C" void gpi_get_log_handler(gpi_log_handler_type **handler, void **userdata)
-{
+extern "C" void gpi_get_log_handler(gpi_log_handler_type **handler,
+                                    void **userdata) {
     *handler = current_handler;
     *userdata = current_userdata;
 }
 
-
-extern "C" void gpi_set_log_handler(gpi_log_handler_type *handler, void *userdata)
-{
+extern "C" void gpi_set_log_handler(gpi_log_handler_type *handler,
+                                    void *userdata) {
     current_handler = handler;
     current_userdata = userdata;
 }
 
-
-extern "C" void gpi_clear_log_handler(void)
-{
+extern "C" void gpi_clear_log_handler(void) {
     current_handler = nullptr;
     current_userdata = nullptr;
 }
 
-
 static int current_native_logger_level = GPIInfo;
 
-
-extern "C" void gpi_native_logger_log(
-    const char *name,
-    int level,
-    const char *pathname,
-    const char *funcname,
-    long lineno,
-    const char *msg,
-    ...)
-{
+extern "C" void gpi_native_logger_log(const char *name, int level,
+                                      const char *pathname,
+                                      const char *funcname, long lineno,
+                                      const char *msg, ...) {
     va_list argp;
     va_start(argp, msg);
     gpi_native_logger_vlog(name, level, pathname, funcname, lineno, msg, argp);
     va_end(argp);
 }
 
-
 // Decode the level into a string matching the Python interpretation
 struct _log_level_table {
     long level;
     const char *levelname;
 };
 
-static struct _log_level_table log_level_table [] = {
-    { 10,       "DEBUG"         },
-    { 20,       "INFO"          },
-    { 30,       "WARNING"       },
-    { 40,       "ERROR"         },
-    { 50,       "CRITICAL"      },
-    { 0,        NULL}
-};
+static struct _log_level_table log_level_table[] = {
+    {10, "DEBUG"}, {20, "INFO"},     {30, "WARNING"},
+    {40, "ERROR"}, {50, "CRITICAL"}, {0, NULL}};
```

Comment:
You can get similar formatting as before by adding a trailing `,`.

Reply:
It moved the closing `};` bracket but kept the 3 per line formatting.